aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -78,44 +78,59 @@ SparseTensor new_with_tensor_sparse(const LongTensor& indices, const Tensor& val
-
   const SparseType& dtype = values.type().toSparse();
 
   // If sizes are not given, it is inferred as max index of each dim.
   int64_t sparseDims = indices.size(0);
   int64_t denseDims = values.dim() - 1;
 
   std::vector<int64_t> computed_sizes(sparseDims + denseDims);
-  // NB: It used to keepdim. I think that was wrong.
-  LongTensor computed_indices_sizes = std::get</* values */ 0>(indices.max(/* dim */ 1, /* keepdim */ false));
-  computed_indices_sizes.add_(1); // len = max_index + 1
-  LongTensor cpu_computed_indices_sizes;
-  if (computed_indices_sizes.is_cuda()) {
-    cpu_computed_indices_sizes = at::CPU(kLong).tensor(computed_indices_sizes.sizes());
-    cpu_computed_indices_sizes.copy_(computed_indices_sizes);
+  if (indices.numel() > 0) {
+    // If the indices has elements in it, we infer the minimum sparse dimension sizes
+    // as the max value of each dim in indices.
+    // NB: It used to keepdim. I think that was wrong.
+    LongTensor computed_indices_sizes = std::get</* values */ 0>(indices.max(/* dim */ 1, /* keepdim */ false));
+    computed_indices_sizes.add_(1); // len = max_index + 1
+    LongTensor cpu_computed_indices_sizes;
+    if (computed_indices_sizes.is_cuda()) {
+      cpu_computed_indices_sizes = at::CPU(kLong).tensor(computed_indices_sizes.sizes());
+      cpu_computed_indices_sizes.copy_(computed_indices_sizes);
+    } else {
+      cpu_computed_indices_sizes = computed_indices_sizes;
+    }
+    auto cpu_computed_indices_sizes_accessor = cpu_computed_indices_sizes.accessor<int64_t, 1>();
+    for (int64_t d = 0; d < sparseDims; d++) {
+      computed_sizes[static_cast<size_t>(d)] = cpu_computed_indices_sizes_accessor[d];
+    }
   } else {
-    cpu_computed_indices_sizes = computed_indices_sizes;
-  }
-  auto cpu_computed_indices_sizes_accessor = cpu_computed_indices_sizes.accessor<int64_t, 1>();
-  for (int64_t d = 0; d < sparseDims; d++) {
-    computed_sizes[static_cast<size_t>(d)] = cpu_computed_indices_sizes_accessor[d];
+    // If the indices doesn't have elements in it, there is not enough information
+    // to know what the minimum sparse dimension sizes should be, and in this case
+    // we set them to 0
+    for (int64_t d = 0; d < sparseDims; d++) {
+      computed_sizes[static_cast<size_t>(d)] = 0;
+    }
   }
   for (int64_t d = 0; d < denseDims; d++) {
     computed_sizes[static_cast<size_t>(sparseDims + d)] = values.size(d+1);
   }
   return _new_with_dims_and_tensor_sparse(dtype, sparseDims, denseDims, computed_sizes, indices, values);
 }
 
-SparseTensor new_with_size_sparse(const SparseType& dtype, ArrayRef<int64_t> size) {
+SparseTensor new_with_dims_and_size_sparse(const SparseType& dtype, int64_t sparseDims, int64_t denseDims, ArrayRef<int64_t> size) {
   SparseTensor self = new_sparse(dtype);
-  _raw_resize_sparse(self, size.size(), 0, size);
+  AT_CHECK(size.size() != 0, "cannot construct sparse tensor with empty size and no values; use the nullary constructor instead");
```

Comment:
what's the nullary constructor?  If you mean e.g. sparse.FloatTensor that doesn't seem applicable because you can't create a scalar.